swarm/src/protocols_handler/one_shot.rs

```diff
@@ -67,7 +69,8 @@ where
     #[inline]
     pub fn new(
         listen_protocol: SubstreamProtocol<TInProto>,
-        inactive_timeout: Duration
+        inactive_timeout: Duration,
+        substream_timeout: Duration
```

Comment:
I saw this PR before you force-pushed to it, and I preferred the way it was before, with a configuration struct!

Could you please extract these two fields?

```suggestion
        config: OneShotHandlerConfig,
```

Reply:
Done!